### DISCUSSION TITLE
Trying to change the inner html of a span to an item on an array, but its not working. I also have the array's items randomized

### DISCUSSION THREAD

**Discussion by DevRenamon-Alder** [community/community#139647] [Programming Help]
### Body

heres the repo https://github.com/DevRenamon-Alder/Test-PartneredDigimon

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Salgado2004** [comment]
Good evening @DevRenamon-Alder 

I had a look in your project and I found the possible cause of the problem.

In the very start you're defining 'clickers' that are DOM elements of digimon by their Id
```javascript
let popup1 = document.getElementById('popup1');
let clicker1 = document.getElementById('digimon-1');
[...]
let popup6 = document.getElementById('popup6');
let clicker6 = document.getElementById('digimon-6');
```

But if you check your HTML, you'll see there's no class with `id="digimon-6"`, resulting in the clicker6 evaluating to null.
You can see that in the browser console:
![image](https://github.com/user-attachments/assets/c2729641-5aca-46c7-975c-0f4466676fcd)
That error is affecting the rest of your code.
When you correctly add the id in the HTML div, it starts to work:
```html
<div class="digimon-6" id="digimon-6">
```
![image](https://github.com/user-attachments/assets/7301cc69-3055-4257-b928-fe60de661b72)

As for the favorite food evaluating to [object HTMLSpanElement], I believe is because of the expression bellow:
```javascript
document.getElementById("favoriteFood1").innerHTML = favoriteFood1;
```
If you change `favoriteFood1` to your favFood array, it should work just fine 😉

Hope you keep working and improving this project! Good Luck 🚀👋 If you liked this answer, please mark as resolved 🙏
